File: M02/Assignment/SprintChallenge3_PokeApi/app/src/main/java/com/lambdaschool/sprintchallenge3_pokeapi/PokemonDao.java
package com.lambdaschool.sprintchallenge3_pokeapi;

import org.json.JSONException;
import org.json.JSONObject;

public class PokemonDao {
    private static final String POKEMON_URL = "https://pokeapi.co/api/v2/pokemon/%s/"; // add pokemon name as

    public static Pokemon getPokemon(String name) {
        Pokemon pokemon;
        try {
            pokemon = new Pokemon(new JSONObject(NetworkAdapter.httpRequest(String.format(POKEMON_URL, name), "GET")));
        } catch (JSONException e) {
            e.printStackTrace();
            pokemon = new Pokemon(name);
        }
        return pokemon;
    }
}

File: M02/Assignment/SprintChallenge3_PokeApi/app/src/main/java/com/lambdaschool/sprintchallenge3_pokeapi/PokemonDao.kt
package com.lambdaschool.sprintchallenge3_pokeapi

import com.lambdaschool.sprintchallenge3_pokeapi.data.source.remote.PokemonRepo
import com.lambdaschool.sprintchallenge3_pokeapi.model.Pokemon
import org.json.JSONException
import org.json.JSONObject

object PokemonDao {
    fun getPokemon(id: Int): Pokemon {
        val pokemonRepo = PokemonRepo()
        var pokemon: Pokemon

        pokemon = pokemonRepo.getPokemonFromAPI(id)!!

        return pokemon
    }
    fun getPokemon(name: String): Pokemon? {
        val pokemonRepo = PokemonRepo()
        var pokemon: Pokemon?

        pokemon = pokemonRepo.getPokemonFromAPI(name)

        return pokemon
    }
}
